x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/index/store/SearchableSnapshotDirectory.java

```diff
@@ -481,7 +482,9 @@ private void prewarmCache(ActionListener<Void> listener) {
                         logger.trace("{} warming cache for [{}] part [{}/{}]", shardId, file.physicalName(), part + 1, numberOfParts);
                         final long startTimeInNanos = statsCurrentTimeNanosSupplier.getAsLong();
                         ((CachedBlobContainerIndexInput) input).prefetchPart(part);
-                        recoveryState.getIndex().addRecoveredBytesToFile(file.physicalName(), file.partBytes(part));
+                        if (fileFullyCached == false) {
```

Comment:
I wonder if we could do things differently within the `prefetchPart()` so that the method updates the recovered bytes ? `prefetchPart()` already detects when the file is already fully prewarmed.